### PR TITLE
JSUI-2316 Improve Category Facet + Breadcrumb accessibility

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2316

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/CategoryFacet/CategoryFacet.ts [hunk 1]
```diff
@@ -626,26 +627,48 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
     return Utils.arrayEqual(this.activePath, this.options.basePath);
   }
 
+  private buildClearIcon() {
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacet.ts]: I'm seeing this file is quite long. I would prefer to extract code into a CategoryFacetHeader.ts file, that would contain the clear button and the title section.

Even better might be to extract the eraser code out of FacetHeader.ts file and convert it to a component that can accept onClick method, logAnalytics method etc. If that is too complicated, then just extracting could be enough for now.

File: src/ui/CategoryFacet/CategoryFacet.ts [hunk 2]
```diff
@@ -626,26 +627,48 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
...
+    SVGDom.addClassToSVGInContainer(clearIcon.el, 'coveo-facet-header-eraser-svg');
+
+    const onClearClick = () => {
+      this.logAnalyticsEvent(analyticsActionCauseList.categoryFacetClear);
+      this.reset();
+    };
+
+    new AccessibleButton()
+      .withElement(clearIcon.el)
+      .withTitle(title)
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacet.ts]: Is adding a title attribute that is the same as the label attribute needed? In my experience, the screen reader ends up reading e.g. "clear button clear". I find "clear button" much nicer.

File: unitTests/ui/BreadcrumbValuesListTest.ts [hunk 3]
```diff
@@ -42,10 +42,13 @@ export function BreadcrumbValueListTest() {
...
+    it('should display a label with the right attributes for each selected values', () => {
       const built = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement).build();
       const values = $$(built).findAll('.coveo-facet-breadcrumb-value');
       expect(values.length).toBe(3);
+      expect($$(values[0]).getAttribute('aria-label')).toBe('Remove filter on foo0');
+      expect($$(values[0]).getAttribute('role')).toBe('button');
+      expect($$(values[0]).getAttribute('tabindex')).toBe('0');
     });
 
     it('should display a caption for each selected values', () => {
```
Comment (samisayegh) [inline unitTests/ui/BreadcrumbValuesListTest.ts]: Should there be some unit tests for the CategoryFacet clear button?

File: src/ui/CategoryFacet/CategoryFacet.ts (hunk 1 above)
Comment (olamothe) [reply src/ui/CategoryFacet/CategoryFacet.ts]: +1

File: src/ui/CategoryFacet/CategoryFacetHeader.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import { AccessibleButton } from '../../utils/AccessibleButton';
+
+export interface ICategoryFacetHeaderOptions {
+  categoryFacet: CategoryFacet;
+  title: string;
+}
+
+export class CategoryFacetHeader {
+  public element: HTMLElement;
+  public eraserElement: HTMLElement;
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacetHeader.ts]: Would it be possible to encapsulate these variables such as `eraserElement` and `waitElement`? Also on line 18, could the `options` variable be private?

File: unitTests/ui/CategoryFacet/CategoryFacetTest.ts [hunk 5]
```diff
@@ -10,25 +10,25 @@ import { IBuildingQueryEventArgs } from '../../../src/events/QueryEvents';
...
-    const queryBuilder = new QueryBuilder();
-    fakeResults.categoryFacets.push(FakeResults.createFakeCategoryFacetResult('@field', [], 'value', numberOfResults));
-    queryBuilder.categoryFacets.push({
-      field: '@field',
-      path: pluck(fakeResults.categoryFacets[0].parentValues, 'value'),
-      maximumNumberOfValues: numberOfRequestedValues
-    });
-    return { results: fakeResults, query: queryBuilder.build() };
-  }
+export function buildSimulateCategoryFacetQueryData(numberOfResults = 11, numberOfRequestedValues = 11): ISimulateQueryData {
```
Comment (samisayegh) [inline unitTests/ui/CategoryFacet/CategoryFacetTest.ts]: Hmm, not sure about the name here. What about `buildCategoryFacetResults` or `createCategoryFacetResults`?

**Pull request merged**